src/main/java/io/quarkus/search/app/SearchService.java

```diff
@@ -111,28 +119,74 @@ public SearchResult<GuideSearchHit> search(@RestQuery @DefaultValue(QuarkusVersi
+                                    .numberOfFragments(contentSnippets)
+                                    .fragmentSize(contentSnippetsLength)
+                                    // The rest of fragment configuration is static
+                                    .orderByScore(true)
+                                    // We don't use sentence boundaries because those can result in huge fragments
+                                    .boundaryScanner().chars().boundaryMaxScan(10).end())
                     .sort(f -> f.score().then().field(language.addSuffix("title_sort")))
                     .routing(QuarkusVersionAndLanguageRoutingBinder.searchKeys(version, language))
                     .totalHitCountThreshold(TOTAL_HIT_COUNT_THRESHOLD + (page + 1) * PAGE_SIZE)
+                    .toQuery();
+
+            hackQuery(query, q, language, highlightCssClass);
+
+            var result = query
+                    .extension(ElasticsearchExtension.get())
+
                     .fetch(page * PAGE_SIZE, PAGE_SIZE);
-            return new SearchResult<>(result);
+
+            return new SearchResult<>(result, extractSuggestion(result));
+        }
+    }
+
+    private String extractSuggestion(ElasticsearchSearchResult<GuideSearchHit> result) {
+        JsonArray options = result.responseBody().getAsJsonObject("suggest")
+                .getAsJsonArray("didYouMean")
+                .get(0).getAsJsonObject()
+                .getAsJsonArray("options");
+        if (options != null && !options.isEmpty()) {
+            return options.get(0).getAsJsonObject().get("highlighted").getAsString();
+        } else {
+            return null;
+        }
+    }
+
+    private void hackQuery(SearchQuery<GuideSearchHit> query, String q, Language language, String highlightCssClass) {
+        try {
+            Field payloadField = ElasticsearchSearchQueryImpl.class.getDeclaredField("payload");
+            payloadField.setAccessible(true);
+            JsonObject payload = (JsonObject) payloadField.get(query);
+            JsonObject suggest = new JsonObject();
+            payload.add("suggest", suggest);
+            suggest.addProperty("text", q);
+            JsonObject suggestion = new JsonObject();
+            suggest.add("didYouMean", suggestion);
+            JsonObject phrase = new JsonObject();
+            suggestion.add("phrase", phrase);
+            phrase.addProperty("field", language.addSuffix("fullContent_suggestion"));
+            phrase.addProperty("size", 1);
+            phrase.addProperty("gram_size", 1);
+            JsonObject highlight = new JsonObject();
+            phrase.add("highlight", highlight);
+            highlight.addProperty("pre_tag", "<span class=\"" + highlightCssClass + "\">");
+            highlight.addProperty("post_tag", "</span>");
+        } catch (NoSuchFieldException | IllegalAccessException e) {
+            throw new RuntimeException(e);
         }
     }
```

Comment:
Why don't you simply use https://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#search-dsl-query-elasticsearch-json to edit the JSON? It doesn't require reflection and it's an actual API :)

Reply:
That's what I was looking for but didn't find! 🙈 😃 thanks! 😃